azurelinuxagent/daemon/main.py

```diff
@@ -99,6 +99,17 @@ def check_pid(self):
 
         fileutil.write_file(pid_file, ustr(os.getpid()))
 
+    def sleep_if_disabled(self):
+        agent_disabled_file_path = conf.get_disable_agent_file_path()
+        if os.path.exists(agent_disabled_file_path):
+            import threading
+            logger.warn("Disabling the guest agent by sleeping forever; "
```

Comment:
There's no API to disable the agent, there's just a file on disk?

I assumed we would just stop the service, and no code would be running.

Reply:
The API is in CRP, exposed through ARM/SDK/CLI. Cloud-init will honor this flag by stopping/disabling the service, but for distros without cloud-init support, I believe this is the best we can do.